### PR TITLE
Fix checkout flow ensure line_items are in stock before every checkou…

### PR DESCRIPTION
**Pull request by tanmay3011**
…t step

### CODE REVIEW

**Comment by priyank-gupta** [comment]
:+1: 
Seems fine to me. @nishant-cyro ?

File: core/spec/models/spree/order/checkout_spec.rb [hunk 1]
```diff
@@ -274,6 +285,17 @@ def assert_state_changed(order, from, to)
             end
           end
         end
+
+        context 'line_items are not in stock' do
```
Comment (nishant-cyro) [inline core/spec/models/spree/order/checkout_spec.rb:293]: @tanmay3011 - Please ensure order should be in address state.

File: core/app/models/spree/order/checkout.rb [hunk 2]
```diff
@@ -103,6 +110,8 @@ def self.define_state_machine!
               end
 
               if states[:delivery]
+                before_transition to: :delivery, do: :ensure_line_item_variants_are_not_discontinued
```
Comment (nishant-cyro) [inline core/app/models/spree/order/checkout.rb:113]: @tanmay3011 - Add specs for ensure_line_item_variants_are_not_discontinued.

File: core/app/models/spree/order.rb [hunk 3]
```diff
@@ -512,7 +512,7 @@ def restart_checkout_flow
         state: 'cart',
         updated_at: Time.current,
       )
-      self.next! if self.line_items.size > 0
+      self.next! if insufficient_stock_lines.blank? && line_items.size > 0
```
Comment (nishant-cyro) [inline core/app/models/spree/order.rb]: @tanmay3011 - Please also ensure line_items shouldn't be discontinued.

Comment (nishant-cyro) [inline core/app/models/spree/order.rb]: @tanmay3011 - Please add specs for these.

**Comment by nishant-cyro** [comment]
Looks good to me. Please create a direct PR on spree :+1:

**Pull request closed without merging**